### PR TITLE
Bvs/oct 25 education gui updates

### PR DESCRIPTION
**Pull request by bschweiz**
### Resolves

_What Github issue does this resolve (please include link)?_

https://github.com/orgs/ComputeCycles/projects/1#card-70903415
https://github.com/orgs/ComputeCycles/projects/1#card-71554601
https://github.com/orgs/ComputeCycles/projects/1#card-71560523
https://github.com/orgs/ComputeCycles/projects/1#card-70917964

### Proposed Changes

_Describe what this Pull Request does_

1. cleanup/ fix variable naming in lighting sequence parse code block for virtual/mirror sats
2. fix error on LS: STOPCLEAR broadcast to virtual/mirror sats 
3. keep VM's internal list of mirrored Sats properly up to date by fixing `removeVirtualSatFromList` function
4. refactor mqttControl.js virtual light sequence parsing code

### Reason for Changes

_Explain why these changes should be made_

### Test Coverage

_Please show how you have added tests to cover your changes_

### CODE REVIEW

File: src/engine/mqttControl.js [hunk 1]
```diff
@@ -433,116 +432,127 @@ class MqttControl extends EventEmitter{
...
-                            const convertedHSV = this.convertHSVBrightness(`#${splitHex[1]}`);
+            while (loopAmount > 0) {
+                let posCopy = true;
+                const positions = ['#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000', '#000000'];
+                for (let i = 0; i < totalPos.length; i++) {
+                    if (posCopy) {
+                        this.positionsCopy = [...positions];
+                    }
+                    let splitHex = '';
+                    if (totalPos[i].includes('L')) {
```
Comment (rvsrvs) [inline src/engine/mqttControl.js:485]: `If` statements that get this long _really_ need to be separated out into function calls.  My general rule is that any JS function more than 10 lines long needs to be broken down.  bc when you break it down, you find cut'n'paste kinds of code you didn't know was there..

**Review by rvsrvs — CHANGES_REQUESTED** [review]
far too monolithic.  let's break this up.

File: src/engine/mqttControl.js (hunk 1 above)
Comment (bschweiz) [reply src/engine/mqttControl.js:485]: broken down into functions now

**Review by bschweiz — COMMENTED** [review]
if statements now call functions, more modular approach

**Comment by rvsrvs** [comment]
Much nicer..  Can go farther down this path, but I'm going to merge this..

**Pull request merged**